types/adobe__ccweb-add-on-sdk/index.d.ts

```diff
@@ -0,0 +1,39 @@
+/********************************************************************************
+ * MIT License
+
+ * © Copyright 2023 Adobe. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ ********************************************************************************/
+
+declare module "https://new.express.adobe.com/static/add-on-sdk/sdk.js" {
```

Comment:
I guess this package isn't real, it's just a URL... I don't think I've seen someone do a package like this before. Are people really expected to import it like this?

Reply:
Yes, it defines a module name with the URL, and re-exports the types from `@adobe/ccweb-add-on-sdk-types` to facilitate the imports from the URL as module.

Sample use:
```
import addOnUISdk from "https://new.express.adobe.com/static/add-on-sdk/sdk.js";
addOnUISdk.ready.then(() => {
    console.log("addOnUISdk is ready for use.");
});
```

In Adobe-express this URL provides APIs to be used in the add-on code.
This is some [documentation](https://developer.adobe.com/express/add-ons/docs/references/addonsdk/)

Reply:
Hi @jakebailey , to provide some more context:

The SDK is utilized by Express Add-Ons within their iframe to establish a connection and interact with the underlying Express web application through the APIs exposed by the SDK.

The iframe employs a static URL link to include the SDK script. This approach is similar to how other application plugin SDKs work. For example, Microsoft Office add-ins must include the "https://appsforoffice.microsoft.com/lib/1/hosted/office.js" URL in their script tag. (Reference: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/develop-overview#access-the-office-javascript-api-library)

Traditionally, these SDKs inject an SDK object into the global space. For instance, the Office SDK injects the Office object under the global window.

However, Adobe Express has taken a different approach. Here, we expect the add-in code to import the SDK to obtain a handle to the SDK object. This decision was made to avoid polluting the global namespace of the add-on and to maintain cleanliness.